ui/src/track-layout/track-layout-model.tsx

```diff
@@ -179,6 +180,43 @@ export function getNameSpecifier(
     return nameStructure && 'specifier' in nameStructure ? nameStructure.specifier : undefined;
 }
 
+export const locationTrackNameFieldsSanitized = (
+    saveRequest: LocationTrackSaveRequest,
+): Pick<LocationTrackSaveRequest, 'nameFreeText' | 'nameSpecifier'> => {
+    const freeText = !isNilOrBlank(saveRequest.nameFreeText)
+        ? saveRequest.nameFreeText?.trim()
+        : undefined;
+    const specifier = saveRequest.nameSpecifier;
+
+    switch (saveRequest.namingScheme) {
+        case LocationTrackNamingScheme.FREE_TEXT:
+        case LocationTrackNamingScheme.WITHIN_OPERATING_POINT:
+            return {
+                nameFreeText: freeText,
+                nameSpecifier: undefined,
+            };
+        case LocationTrackNamingScheme.TRACK_NUMBER_TRACK:
+            return {
+                nameFreeText: freeText,
+                nameSpecifier: specifier,
+            };
+        case LocationTrackNamingScheme.BETWEEN_OPERATING_POINTS:
+            return {
+                nameFreeText: undefined,
+                nameSpecifier: specifier,
+            };
+        case LocationTrackNamingScheme.CHORD:
+            return {
+                nameFreeText: undefined,
+                nameSpecifier: undefined,
+            };
+        case undefined:
+            throw Error('Naming scheme is mandatory for location track save request!');
```

Comment:
Tässä sama kuin toisaalla, `LocationTrackSaveRequest` sallii sen että nimeämisskeema on `undefined` vaikka näin käytännössä koskaan ei tulekaan tallennusvaiheessa enää olemaan. Tämäkin tilanne piti siis hanskata, joten heitetään siitä vaan virhe